samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -800,8 +800,9 @@ class SamzaContainer(
 
       val startTime = System.nanoTime()
       status = SamzaContainerStatus.STARTING
-
-      jmxServer = new JmxServer()
+      if (new JobConfig(config).getJMXEnabled) {
```

Comment:
```suggestion
      if (config.getJMXEnabled) {
```

Reply:
I will prefer this

Reply:
> I will prefer this, 

I think each of our config class in scala defines implicit conversion methods. For instance, JobConfig.scala defines implicit conversion [here](https://github.com/apache/samza/blob/7780081ea147ee2854a982e020f9a5f6e0ab9fc8/samza-core/src/main/scala/org/apache/samza/config/JobConfig.scala#L104). Can you please share the rationale why you prefer explicitly creating object  here?

Reply:
>> I will prefer this

I think each of our config class in scala defines implicit conversion methods. For instance, JobConfig.scala defines implicit conversion [here](https://github.com/apache/samza/blob/7780081ea147ee2854a982e020f9a5f6e0ab9fc8/samza-core/src/main/scala/org/apache/samza/config/JobConfig.scala#L104). Can you please share the rationale why you prefer explicitly creating object here?

Reply:
This will ensure, method is from the JobConfig subclass of Config and not any other subclass

Reply:
Majority of the implementation in `SamzaContainer` involving scala code follows this scala implicit conversion and does not create explicit object here. If I understand right, you're saying we will have different getJmxEnabled from other config class. Can you share the scenarios these can happen(to me at-least the reason sounds far-fetched).